execution_workflows/IsoSCM/modules/bam_to_fastq.nf

```diff
@@ -0,0 +1,37 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+def options    = initOptions(params.options)
+
+/*
+    Convert input BAM file to fastq for star alignment step
+*/
+process BAM_TO_FASTQ {
+    container "docker.io/apaeval/isoscm:latest"
+
+    input:
+    tuple val(sample), path(bam), val(read_type)
+
+    output:
+    tuple val(sample), path(fastq1), path(fastq2), emit: ch_fastq_files
+
+    script:
+    sorted_bam = sample + ".qsort.bam"
+    fastq1 = sample + ".fastq1.fastq"
+    fastq2 = sample + ".fastq2.fastq"
+    if (read_type == "paired") {
+    	"""
+    	samtools sort -n -o $sorted_bam $bam
+        bedtools bamtofastq -i $sorted_bam \
+                            -fq $fastq1 \
+                            -fq2 $fastq2
+    	"""
+    }
+    else {
+        """
+        bedtools bamtofastq -i $bam -fq $fastq1
+        cp $fastq1 $fastq2
+        """
+    }
```

Comment:
Is the `cp $fastq1 $fastq2` needed for pipeline plumbing? If so happy for it to stay, otherwise it feels a bit unnecessary to  duplicate the file (which could be in the GB range) if STAR doesn't use it? If for pipeline plumbing (i.e. nextflow needs to know file has been generated) then maybe could just `touch $fastq2` to save space?

Also, do we need the fastqs to be uncompressed? Again it's not a critical change, but would save on disk space requirements for runs on full datasets. For STAR you can also provide the `--readFilesCommand zcat` or `--readFilesCommand gunzip -c` so it knows to decompress the input files.